samza-core/src/test/java/org/apache/samza/execution/TestProcessorGraph.java

```diff
@@ -194,5 +203,9 @@ public void testTopologicalSort() {
     assertTrue(idxMap2.get("6") > idxMap2.get("1"));
     assertTrue(idxMap2.get("5") > idxMap2.get("4"));
     assertTrue(idxMap2.get("7") > idxMap2.get("5"));
+
+    //test graph3
+    List<ProcessorNode> sortedNodes3 = graph3.topologicalSort();
```

Comment:
I ran the unit test without this "if" condition and it worked as expected. The "if" was added later for optimization.